File: src/main/java/com/devahn/app/api/ApiResponse.java
package com.devahn.app.api;

import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor
public class ApiResponse<T> {
    private boolean success;
    private T data;
    private String message;
    private String errorCode;
    private long timestamp;

    public ApiResponse(boolean success, T data, String message, String errorCode) {
        this.success = success;
        this.data = data;
        this.message = message;
        this.errorCode = errorCode;
        this.timestamp = System.currentTimeMillis();
    }

    public static <T> ApiResponse<T> ok(T data) {
        return new ApiResponse<>(true, data, "Success", null);
    }

    public static <T> ApiResponse<T> ok(T data, String message) {
        return new ApiResponse<>(true, data, message, null);
    }

    public static <T> ApiResponse<T> error(String errorCode, String message) {
        return new ApiResponse<>(false, null, message, errorCode);
    }
}

File: WebContent/app/BE/backend/utils/api-response.ts
// ============================================================
// ApiResponse — Spring Boot ApiResponse<T> 완전 이식
// 모든 API Route에서 일관된 응답 형식을 보장
// ============================================================

export interface ApiResponse<T> {
  success: boolean;
  data: T | null;
  message: string;
  errorCode: string | null;
  timestamp: number;
}

export function ok<T>(data: T, message = "Success"): ApiResponse<T> {
  return { success: true, data, message, errorCode: null, timestamp: Date.now() };
}

export function error(errorCode: string, message: string): ApiResponse<null> {
  return { success: false, data: null, message, errorCode, timestamp: Date.now() };
}
